### PR TITLE
CKS CoreOS EOL update 

### PR DESCRIPTION
**Pull request by Pearl1594**
## Description
With CoreOS having reached End of life, support has been provided CKS to consume systemVM template to deploy kubernetes clusters

## Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] Enhancement (improves an existing feature and functionality)
- [ ] Cleanup (Code refactoring and cleanup, that may add test cases)

### CODE REVIEW

**Comment by Pearl1594** [comment]
@blueorangutan package

**Comment by Pearl1594** [comment]
@blueorangutan package

File: engine/schema/src/main/resources/META-INF/db/schema-41400to41500.sql [hunk 1]
```diff
@@ -833,3 +833,7 @@ INSERT INTO `cloud`.`guest_os_hypervisor` (uuid,hypervisor_type, hypervisor_vers
 
 -- Fix OS category for Guest OS 'Other PV Virtio-SCSI (64-bit)'
 UPDATE `cloud`.`guest_os` SET category_id = 7 WHERE id = 275 AND display_name = 'Other PV Virtio-SCSI (64-bit)';
+
+ALTER TABLE `cloud`.`user_vm` ADD COLUMN `user_vm_type` varchar(255) DEFAULT "UserVM" COMMENT 'Defines the type of UserVM';
```
Comment (sureshanaparti) [inline engine/schema/src/main/resources/META-INF/db/schema-41400to41500.sql:837]: @Pearl1594 is this not redundant of _Type_ column in _vm_instance_ table? This _type_ column already used to persist different system VMs  (ConsoleProxy, SecondaryStorageVm, DomainRouter), may be you can use this for different user VMs as well.

File: engine/schema/src/main/java/com/cloud/vm/UserVmVO.java [hunk 2]
```diff
@@ -48,6 +48,9 @@
     @Column(name = "update_parameters", updatable = true)
     protected boolean updateParameters = true;
 
+    @Column(name = "user_vm_type", updatable = true)
```
Comment (sureshanaparti) [inline engine/schema/src/main/java/com/cloud/vm/UserVmVO.java:51]: can use "type" from VMInstanceVO instead?

File: engine/orchestration/src/main/java/org/apache/cloudstack/engine/orchestration/VolumeOrchestrator.java [hunk 3]
```diff
@@ -824,7 +824,7 @@ private DiskProfile allocateTemplatedVolume(Type type, String name, DiskOffering
         int volumesNumber = 1;
         List<DatadiskTO> templateAsIsDisks = null;
         String configurationId = null;
-        if (template.isDeployAsIs()) {
+        if (template.isDeployAsIs() && vm.getType() != VirtualMachine.Type.SecondaryStorageVm) {
```
Comment (sureshanaparti) [inline engine/orchestration/src/main/java/org/apache/cloudstack/engine/orchestration/VolumeOrchestrator.java:827]: please note there is already a specific PR #4437 with this change

File: engine/schema/src/main/resources/META-INF/db/schema-41400to41500.sql (hunk 1 above)
Comment (Pearl1594) [reply engine/schema/src/main/resources/META-INF/db/schema-41400to41500.sql:837]: Intention of adding a user vm type - as a sub-type of VMInstance type - User is to basically reduce any regression - changing User to other type to accommodate CKS / K8s nodes would possibly require a lot more changes

File: engine/orchestration/src/main/java/org/apache/cloudstack/engine/orchestration/VolumeOrchestrator.java (hunk 3 above)
Comment (Pearl1594) [reply engine/orchestration/src/main/java/org/apache/cloudstack/engine/orchestration/VolumeOrchestrator.java:827]: This was added to verify the fix @sureshanaparti

File: server/src/main/java/com/cloud/vm/UserVmManager.java [hunk 4]
```diff
@@ -55,6 +55,10 @@
 
     static final int MAX_USER_DATA_LENGTH_BYTES = 2048;
 
+    public static enum UserVmType {
```
Comment (sureshanaparti) [inline server/src/main/java/com/cloud/vm/UserVmManager.java:58]: @Pearl1594  there is a common enum "VirtualMachine.Type" where all the VM types are defined, can use the same?

File: engine/schema/src/main/resources/META-INF/db/schema-41400to41500.sql (hunk 1 above)
Comment (sureshanaparti) [reply engine/schema/src/main/resources/META-INF/db/schema-41400to41500.sql:837]: Ok @Pearl1594 I think it is better to use single source for defining the instance type.

**Comment by Pearl1594** [comment]
@blueorangutan package

**Pull request closed without merging**